cfgov/unprocessed/js/modules/ClearableInput.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// Required polyfills for IE9.
+if ( !Modernizr.classlist ) { require( '../modules/polyfill/class-list' ); } // eslint-disable-line no-undef, global-require, no-inline-comments, max-len
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * ClearableInput
+ * @class
+ *
+ * @classdesc Initializes a new ClearableInput molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} A ClearableInput instance.
+ */
+function ClearableInput( element ) {
+  var BASE_CLASS = 'input-contains-label';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'ClearableInput' );
+  var _inputDom = _dom.querySelector( 'input' );
+  var _clearBtnDom = _dom.querySelector( '.' + BASE_CLASS + '_after__clear' );
+
+  /**
+   * @returns {Object} The ClearableInput instance.
+   */
+  function init() {
+    _clearBtnDom.addEventListener( 'mousedown', _clearClicked );
+    _inputDom.addEventListener( 'keyup', _inputTyped );
```

Comment:
Say there's a `t` in the input field, the user could either press any key (e.g. `e`) or they may press the delete key. If it's any other key, it's true that `show` will be called again, but if it's the delete key hide would need to be called immediately to hide the clear button when there are no letters. I don't see how a timeout wouldn't introduce scenarios where it would be possible to type fast enough to hit the delete key and have it be ignored. We could have logic that handles the keyup event differently when there's more than one character in the input, but it would need to handle scenarios where the text was selected and delete was hit.